src/main/java/com/openlattice/datastore/services/EdmService.java

```diff
@@ -547,6 +535,19 @@ private void setupDefaultEntitySetPropertyMetadata( UUID entitySetId, UUID entit
         } );
     }
 
+    private void setupDefaultEntityTypePropertyMetadata( UUID entityTypeId ) {
+        final var et = getEntityType( entityTypeId );
+        et.getProperties().forEach( propertyTypeId -> {
+            EntityTypePropertyKey key = new EntityTypePropertyKey( entityTypeId, propertyTypeId );
+            PropertyType property = getPropertyType( propertyTypeId );
```

Comment:
Reads from hazelcast are expensive and `getPropertyType` reads from hazelcast, so it would be faster to bulk the hazelcast calls by calling `propertyTypes.getAll(et.getProperties())` before the forEach loop, and then referencing that map